Templates/GitlabCIPipeline/.gitlab-ci.yml

```diff
@@ -0,0 +1,550 @@
+        echo [INFO] Verbose = ${verbose}
+        # Git clone
+        zowe rse issue unix ". ~/.bash_profile && gitClone.sh -w ${uniqueWorkspaceId} -r ${CI_REPOSITORY_URL} -b ${CI_COMMIT_REF_NAME} -a ${application}" --user ${RSEAPI_USER} --password ${RSEAPI_PASSWORD} --cwd ${RSEAPI_WORKING_DIRECTORY}
+
+1-Build application:
+    needs: ["Clone"]
+    stage: Build
+    tags: [shell]
+    rules:
+        - if: $CI_COMMIT_TAG
+          when: never
+        - when: on_success
+    dependencies: []
+    artifacts:
+        name: "report-${CI_PIPELINE_ID}"
+        when: always
+        paths:
+            - "$CI_PROJECT_DIR/BUILD-OUTPUT/*.xml"
+        reports:
+            junit: "$CI_PROJECT_DIR/BUILD-OUTPUT/*.xml"
+    script:
+        - zowe rse issue unix ". ~/.bash_profile && env|sort && dbbBuild.sh -w ${uniqueWorkspaceId} -a ${application} -b ${CI_COMMIT_REF_NAME} -p ${pipelineType} ${verbose}" --user ${RSEAPI_USER} --password ${RSEAPI_PASSWORD} --cwd ${RSEAPI_WORKING_DIRECTORY}
+
+2-Prepare and publish build logs:
+    needs: ["1-Build application"]
+    stage: Build
+    tags: [shell]
+    rules:
+        - if: $CI_COMMIT_TAG
+          when: never 
+        - when: on_success
+    dependencies: []
+    variables:
+        sourceLogsFile: ${uniqueWorkspaceId}/logs.tar
+        destinationLogsFile: ${buildOutputDirectory}/logs.tar
+    artifacts:
+        name: "buildReport-${CI_PIPELINE_ID}"
+        when: always
+        paths:
+            - "${buildOutputDirectory}/*.log"
+            - "${buildOutputDirectory}/*.html"
+            - "${buildOutputDirectory}/*.json"
+            - "${buildOutputDirectory}/*.xml"
+            #- "${buildOutputDirectory}/*.pdf"
+            #- "${buildOutputDirectory}/*.cczip"
+            - "${buildOutputDirectory}/*.txt"
+            #- "${buildOutputDirectory}/*SONARQUBE/*.*"
+            # "${buildOutputDirectory}/logs.tar"
+    script: |
+        zowe rse issue unix ". ~/.bash_profile && prepareLogs.sh -w ${uniqueWorkspaceId}" --user ${RSEAPI_USER} --password ${RSEAPI_PASSWORD} --cwd ${RSEAPI_WORKING_DIRECTORY}
+        echo START TRANSFER LOGS FILE = ${sourceLogsFile} to ${destinationLogsFile}
+        if [ ! -z "${sourceLogsFile}" ]; then
+            if [ -d "${buildOutputDirectory}/" ]; then
+                rm -R ${buildOutputDirectory}
+                echo [INFO] Removed existing directory ${buildOutputDirectory}
+            fi
+            # Create directory to store build output
+            mkdir ${buildOutputDirectory}
+            echo [INFO] Created build output directory = ${buildOutputDirectory}
+            zowe rse dl uf "${sourceLogsFile}" -f "${destinationLogsFile}" -b --user ${RSEAPI_USER} --password ${RSEAPI_PASSWORD}
+            tar -xvf "${destinationLogsFile}" -C "${buildOutputDirectory}/" --strip-components 1
+            echo [INFO] Extracted log.tar file to ${buildOutputDirectory}
+        else
+            echo [WARNING] Tar file containing the logs was not found
```

Comment:
my bad, I meant to check if the file exist. I updated to the following.
if [ -f "${sourceLogsFile}" ]; then

Reply:
Oh!! I cannot check this from the definition file since the files are on USS. I would to assume it is created if the script successfully execute. With the updated ZOWE CI, we should be able to trust the RC returned.

Reply:
Updated to not checking any condition since the "sourceLogsFile" will always be defined and assigned the path.